### ISSUE TITLE
Some requirements described in the PS6 are for R rather than for Python and they cause some problems.

### ISSUE THREAD

**Issue by khan1792** [UC-MACSS/persp-model_W18#229]
Some functions described in the assignment is for R rather than for Python. So the requirements become obscure when we use Python because R and Python have different default function setting.

For the part a), if we don't set any control in the first tree in Python, the tree will become incredibly large although the requirement in R is not a problem because of its better default setting (small tree).

For the part b),  "nobs = # number of rows in the training set" and "mindev = 0" are the codes in R. It seems that we cannot set mindev in Python, and I'm also wondering if the "nobs" in R is the same as the "max_leaf_nodes" in Python.

Finally, for the part d), is "mtry = N." (the number of variables) in R the same as "max_features = N." in Python?

**Comment by jgdenby** [comment]
For d), I think what you posted is correct. Here's an excerpt from the [scikit-learn documentation](http://scikit-learn.org/dev/modules/ensemble.html#forest):

 > As other classifiers, forest classifiers have to be fitted with two arrays: a sparse or dense array X of size `[n_samples, n_features]` holding the training samples, and an array Y of size `[n_samples]` holding the target values (class labels) for the training samples:

**Comment by ruixue-li** [comment]
nobs in R is just the number of observation right? Since we're just using the training set as the input to the model, I don't think we need an equivalent to nobs? I'm unsure about part a) and b)..

**Comment by khan1792** [comment]
@ruixue-li  I also finished a version in R ... I think the goal of the part (a) and (b) is to let us build a small tree (part a), a large tree, and a pruned tree (from the large tree) with medium size (part b). At least in R, I can achieve this goal by following the requirements in part a and b. But in python, the results are weird because we will have a large tree in the first part as well.

**Comment by jgdenby** [comment]
has anyone managed to work out how to do part (b) in Python? I'm not totally sure from the documentation what parameters are meant to match up with the R code. Also, how do we conduct the pruning? Do we need to implement the function from the notebook? If so, I'm not super confident in how to do that.

**Comment by ruixue-li** [comment]
For part (b), I used a loop to fit a tree from max depth 1 to max depth 10, and plotted the MSE so that I can find one with smallest MSE. I used the min_sample_leaf = 5 since it seems to me that it matches the controls in R....a little bit.

**Comment by CooperNederhood** [comment]
Yeah, I tried to be clever and minimize mse as function of leaf size and depth to approximate a pruning of some sort but it's not working :(
